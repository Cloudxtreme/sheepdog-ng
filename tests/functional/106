#!/bin/bash

# Test some dog comannds with auto-recovery disabled

. ./common

for i in `seq 0 2`; do
    _start_sheep $i
done

_wait_for_sheep 3

_cluster_format -c 2 -s

$DOG vdi create test 100M

_random | $DOG vdi write test

_kill_sheep 1
_wait_for_sheep_stop 1

$DOG cluster info | _filter_cluster_info
$DOG node list
$DOG node recovery

_random | $DOG vdi write test

_start_sheep 1
_wait_for_sheep 3
_start_sheep 3
sleep 1

$DOG cluster info | _filter_cluster_info
$DOG node list
$DOG cluster info -p 7003

_kill_sheep 1
_wait_for_sheep_stop 1
rm $STORE/1 -rf
_start_sheep 1
sleep 1
$DOG cluster info -p 7001 | _filter_cluster_info

$DOG cluster reconfig
_wait_for_sheep_recovery 0
$DOG cluster info | _filter_cluster_info
$DOG node list

$DOG cluster shutdown
_wait_for_sheep_stop

for i in `seq 0 3`; do
    _start_sheep $i
done

_wait_for_sheep 4

$DOG node recovery
$DOG cluster info | _filter_cluster_info
$DOG node list
