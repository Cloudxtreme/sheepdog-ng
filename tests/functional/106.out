QA output created by 106
using backend plain store
Cluster status: running, auto-recovery disabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2

Offline: [127.0.0.1:7001, ]
Nodes In Recovery:
  Id   Host:Port         V-Nodes       Zone       Progress
Cluster status: running, auto-recovery disabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
127.0.0.1:7003 not yet join the cluster. You may try 'dog cluster reconfig' to manually add it.
Cluster status: running, auto-recovery disabled

Cluster created at DATE

Epoch Time           Version
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
Cluster status: running, auto-recovery disabled

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
   3   127.0.0.1:7003      	127          3
Nodes In Recovery:
  Id   Host:Port         V-Nodes       Zone       Progress
Cluster status: running, auto-recovery disabled

Cluster created at DATE

Epoch Time           Version
DATE      2 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002, 127.0.0.1:7003]
DATE      1 [127.0.0.1:7000, 127.0.0.1:7001, 127.0.0.1:7002]
  Id   Host:Port         V-Nodes       Zone
   0   127.0.0.1:7000      	128          0
   1   127.0.0.1:7001      	128          1
   2   127.0.0.1:7002      	128          2
   3   127.0.0.1:7003      	127          3
